src/editor-service/editor-server.ts

```diff
@@ -21,74 +22,78 @@ import {LocalEditorService} from './local-editor-service';
-  }
+  private _editorService: LocalEditorService;
 
   async handleMessage(message: Request): Promise<any> {
+    if (message.kind === 'init') {
+      if (this._editorService) {
+        throw new Error('Already initialized!');
+      }
+      this._editorService = new LocalEditorService({
+        urlLoader: new FSUrlLoader(message.basedir),
+        urlResolver: new PackageUrlResolver()
+      });
+      return;
+    }
+    const editorService = this._editorService;
+    if (!editorService) {
+      throw new Error(
+          `Must send an 'init' message before any others. ` +
+          `Received '${message.kind}' message before 'init'.`);
+    }
     switch (message.kind) {
       case 'getWarningsFor':
-        return this._localEditorService.getWarningsForFile(message.localPath);
+        return editorService.getWarningsForFile(message.localPath);
       case 'fileChanged':
-        await this._localEditorService.fileChanged(
-            message.localPath, message.contents);
+        await editorService.fileChanged(message.localPath, message.contents);
         return;
-      case 'init':
-        throw new Error('Already initialized!');
       case 'getDefinitionFor':
-        return this._localEditorService.getDefinitionForFeatureAtPosition(
+        return editorService.getDefinitionForFeatureAtPosition(
             message.localPath, message.position);
       case 'getDocumentationFor':
-        return this._localEditorService.getDocumentationAtPosition(
+        return editorService.getDocumentationAtPosition(
             message.localPath, message.position);
       case 'getTypeaheadCompletionsFor':
-        return this._localEditorService.getTypeaheadCompletionsAtPosition(
+        return editorService.getTypeaheadCompletionsAtPosition(
             message.localPath, message.position);
       case '_clearCaches':
-        return this._localEditorService._clearCaches();
+        return editorService._clearCaches();
       default:
-        // This assignment makes it a type error if we don't handle all possible
-        // values of `message.kind`.
         const never: never = message;
         throw new Error(`Got unknown kind of message: ${util.inspect(never)}`);
     }
   }
 }
 
-let server: EditorServer;
-process.once('message', (initRequest: RequestWrapper) => {
-  if (initRequest.value.kind !== 'init') {
-    process.send(<ResponseWrapper>{
-      id: initRequest.id,
-      value: {
-        kind: 'rejection',
-        rejection: `Expected first message to be 'init', ` +
-            `got ${initRequest.value.kind}`
-      }
-    });
-    return;
-  }
-  server = new EditorServer(initRequest.value.basedir);
 
-  process.on('message', async(request: RequestWrapper) => {
-    const result = await getSettledValue(request.value);
-    process.send(<ResponseWrapper>{id: request.id, value: result});
-  });
+const server: EditorServer = new EditorServer();
+
+// stdin/stdout interface
+process.stdin.setEncoding('utf8');
```

Comment:
can this logic be wrapped in a function or process event listener? Right now it looks like there are side effects to just loading this file which we generally want to avoid